lib/s3.rb

```diff
@@ -20,11 +20,19 @@ def save_file(absolute_path_of_file)
             _validate_path(absolute_path_of_file)
             file_name = File.basename(absolute_path_of_file)
             destination_name = create_s3_name(file_name)
-            @bucket.object(destination_name).upload_file(absolute_path_of_file)
-            @bucket.object(destination_name).wait_until_exists
-            _verify_upload(destination_name, absolute_path_of_file)
-            # @bucket.object(s3_name).presigned_url(:get, expires_in: 3600) #uncomment this if public url ends up not working out OPREQ-83850
-            @bucket.object(destination_name).public_url
+            begin
+              @bucket.object(destination_name).upload_file(absolute_path_of_file)
+              @bucket.object(destination_name).wait_until_exists
+              _verify_upload(destination_name, absolute_path_of_file)
+              # @bucket.object(s3_name).presigned_url(:get, expires_in: 3600) #uncomment this if public url ends up not working out OPREQ-83850
+              return @bucket.object(destination_name).public_url
+            rescue Aws::S3::Errors::InvalidAccessKeyId
```

Comment:
For the errors I've encountered, I don't think retry will work. 
The TCP error is happening when we're already waiting for the bucket, which has retry logic baked into it already.
And of course for bad credentials, they stay bad.